File: multilibrary/src/main/java/com/zhongjh/albumcamerarecorder/camera/entity/BitmapData.java
package com.zhongjh.albumcamerarecorder.camera.entity;


/**
 * 拍照制造出来的数据源
 *
 * @author zhongjh
 */
public class BitmapData {

    /**
     * 临时id
     */
    private Long temporaryId;
    /**
     * uri路径
     */
    private String uri;
    /**
     * 真实路径
     */
    private String absolutePath;

    public BitmapData(Long temporaryId, String uri, String absolutePath) {
        this.temporaryId = temporaryId;
        this.uri = uri;
        this.absolutePath = absolutePath;
    }

    public Long getTemporaryId() {
        return temporaryId;
    }

    public void setTemporaryId(Long temporaryId) {
        this.temporaryId = temporaryId;
    }

    public String getUri() {
        return uri;
    }

    public void setUri(String uri) {
        this.uri = uri;
    }

    public String getAbsolutePath() {
        return absolutePath;
    }

    public void setAbsolutePath(String absolutePath) {
        this.absolutePath = absolutePath;
    }


}

File: multilibrary/src/main/java/com/zhongjh/albumcamerarecorder/camera/entity/BitmapData.kt
package com.zhongjh.albumcamerarecorder.camera.entity

import com.zhongjh.albumcamerarecorder.album.entity.Album2


/**
 * 拍照制造出来的数据源
 *
 * @author zhongjh
 */
class BitmapData(
    /**
     * 临时id
     */
    var temporaryId: Long,
    /**
     * uri路径
     */
    var uri: String,
    /**
     * 真实路径
     */
    var absolutePath: String
) {

    /**
     * 用于 DiffUtil.Callback 进行判断
     */
    fun equalsBitmapData(bitmapData: BitmapData): Boolean {
        if (temporaryId != bitmapData.temporaryId) {
            return false
        }

        if (uri != bitmapData.uri) {
            return false
        }
        if (absolutePath != bitmapData.absolutePath) {
            return false
        }
        return true
    }


}
